File: code/1021/1021.rb
n = gets.to_f

def print_values(number_to_print, value_to_print)
  if value_to_print > 1
    puts format('%d nota(s) de R$ %.2f', number_to_print, value_to_print)
  else
    puts format('%d moeda(s) de R$ %.2f', number_to_print, value_to_print)
  end
end

def print_and_return_calc(quant, value)
  number_of_coins = (quant.round(2) / value).to_i
  print_values(number_of_coins, value)
  quant - number_of_coins * value
end

puts 'NOTAS:'
n = print_and_return_calc(n, 100)
n = print_and_return_calc(n, 50)
n = print_and_return_calc(n, 20)
n = print_and_return_calc(n, 10)
n = print_and_return_calc(n, 5)
n = print_and_return_calc(n, 2)
puts 'MOEDAS:'
n = print_and_return_calc(n, 1)
n = print_and_return_calc(n, 0.5)
n = print_and_return_calc(n, 0.25)
n = print_and_return_calc(n, 0.10)
n = print_and_return_calc(n, 0.05)
print_and_return_calc(n, 0.01)

File: code/1021/1021.js
const input = require("fs").readFileSync("/dev/stdin", "utf8");
const lines = input.split("\n");

const printAndCalcBanknotes = (quant, value) => {
  const actualQuant = Math.floor(quant / value);
  console.log(`${actualQuant} nota(s) de R\$ ${value}.00`);
  return quant - actualQuant * value;
};

const printAndCalcCoins = (quant, coins) => {
  const actualQuant = Math.floor(quant.toFixed(2) / coins);
  console.log(`${actualQuant} moeda(s) de R\$ ${coins.toFixed(2)}`);
  return quant - (actualQuant * coins);
};

let n = parseFloat(lines[0]);

console.log("NOTAS:");

n = printAndCalcBanknotes(n, 100);
n = printAndCalcBanknotes(n, 50);
n = printAndCalcBanknotes(n, 20);
n = printAndCalcBanknotes(n, 10);
n = printAndCalcBanknotes(n, 5);
n = printAndCalcBanknotes(n, 2);


console.log("MOEDAS:")
n = printAndCalcCoins(n, 1.00);
n = printAndCalcCoins(n, 0.50);
n = printAndCalcCoins(n, 0.25);
n = printAndCalcCoins(n, 0.10);
n = printAndCalcCoins(n, 0.05);
printAndCalcCoins(n, 0.01);
